_posts/2019-01-08-data-driven-static-resume.adoc

```diff
@@ -0,0 +1,202 @@
+If you also earn a crust from setting up documentation environments for engineering teams and the like, consider building your professional documentation from a single source to show off your skills.
+
+With its singular source and dual link:https://www.codewriting.org/brian-dominick-resume[HTML] and link:https://www.codewriting.org/assets/files/brian-dominick-resume.pdf[PDF] output, my resumé is a simple model implementation of the AJYL tech stack, consisting of *AsciiDoc*, *Jekyll*/*JAMstack*, *YAML*, and *Liquid*.
+
+In this post, I will walk you through how I set it all up, explaining how each component language and utility come into play.
+
+toc::[]
+
+== The Sources
+
+These two output versions (link:https://www.codewriting.org/brian-dominick-resume[HTML] | link:https://www.codewriting.org/assets/files/brian-dominick-resume.pdf[PDF]) are sourced from the exact same files, so they are always in sync;
+I do not have to remember to update two documents whenever I make a change.
+
+Most of the specific content items are stored as _small data_ objects arranged in a link:https://idratherbewriting.com/documentation-theme-jekyll/mydoc_yaml_tutorial.html[YAML] file, similarly to fields in a database.
+A small-data file is simply a text document containing keys and values structured such that both people and machines can modify and consume the organized information.
+
+Here is a YAML snippet drawn directly from the link:https://github.com/briandominick/codewriting/blob/master/data/dominick.yml[single-sourced file].
+
+.From data/dominick.yml -- Resumé small data source file snippet
+[source,yaml]
+----
+include::data/dominick.yml[tags="example-snippet"]
+----
+
+If you inspect the link:https://github.com/briandominick/codewriting/blob/master/data/dominick.yml[whole file], you will see it contains several clusters of data, each analogous to a table in a database.
+
+[TIP]
+One advantage of maintaining small data in flat files is that you can build an entire, perfectly portable datasource with a text editor in mere minutes.
+
+In this case, parameter values contain link:https://asciidoctor.org/docs/what-is-asciidoc/[AsciiDoc] markup.
+Notice on the first line beginning with a `destination:` key, the value text contains AsciiDoc hyperlink markup.
+
+This data gets parsed with a link:https://github.com/Shopify/liquid/wiki[Liquid] template.
+This particular Liquid template (link:https://raw.githubusercontent.com/briandominick/codewriting/master/_templates/liquid/resume.asciidoc[see complete source]) weaves those AsciiDoc-formatted strings from the data file with more AsciiDoc markup.
+
+.From resume.asciidoc -- Template mixes Liquid with AsciiDoc, generates new AsciiDoc source
+[source]
+----
+include::assets/includes/resume.asciidoc[tags="employment"]
+----
+
+Because Liquid templates are usually a mix of at least two markup languages (Liquid markup itself plus your target format), they can get pretty ugly.
```

Comment:
```suggestion
Because Liquid templates are usually a mix of at least two markup languages (Liquid markup itself plus your target format), they can feel pretty busy.
```
Or something like this.... the word 'ugly' kind of feels harsh here.

Reply:
Heh, it's my own code, but point taken ;-).